src/share/poudriere/common.sh

```diff
@@ -7371,6 +7394,10 @@ prepare_ports() {
 		fi
 	fi
 
+	if [ -n "${PACKAGE_BRANCH}" ]; then
+		download_from_repo
+	fi
```

Comment:
1. Need this to not run for non-build commands (like `pkgclean`) and to not run when resuming, and to still respect `-c`  and `-C`. So I think it makes sense higher up (Github review won't let me comment on untouched lines.) :
```
if [ -n "${PACKAGE_BRANCH}" -a "${resuming_build}" -ne 1 ]; then
		download_from_repo
fi
# existing code
if [ ${JAIL_NEEDS_CLEAN} -eq 1 ]; then
			msg_n "Cleaning all packages due to newer version of the jail..."
			rm -rf ${PACKAGES}/* ${cache_dir}
```

2. Then there is the question of what should `-n` (dry run) do? Should it fetch so that a proper accounting of what will build is done or should it just say "will try fetching pkgA pkgB ..."? I have no opinion here.